test/Test/Cardano/Prelude/Helpers.hs

```diff
@@ -0,0 +1,24 @@
+module Test.Cardano.Prelude.Helpers
+  ( eitherLeftBuildable
+  , eitherRightBuildable
+  )
+where
+
+import Cardano.Prelude
+
+import Formatting (Buildable, build, sformat)
+
+import Hedgehog (MonadTest, success)
+import Hedgehog.Internal.Property (failWith)
+
+eitherLeftBuildable
+  :: (MonadTest m, Buildable c) => (a -> Either b c) -> a -> m ()
+eitherLeftBuildable func val = case func val of
+  Left  _   -> success
+  Right res -> failWith Nothing (show $ sformat build res)
+
+eitherRightBuildable
+  :: (MonadTest m, Buildable b) => (a -> Either b c) -> a -> m ()
+eitherRightBuildable func val = case func val of
+  Left  err -> failWith Nothing (show $ sformat build err)
+  Right _   -> success
```

Comment:
Can I nit-pick on the name, and suggest `assertEitherIsRight` and `assertEitherIsLeft`? Seems a bit more clear what the function is asserting. The `Buildable` constraint will show up in the type, so I think we can drop it from the name so things are shorter.

Reply:
That's a way better name. :100: